ananse/network.py

```diff
@@ -623,178 +299,119 @@ def create_promoter_network(self, ddf, prom, fin_expression, outfile, column="tp
-        expression=expression.rename(columns={"index":"gene", "tpm":"target_expression"})
-        
-        package_dir = os.path.dirname(ananse.__file__)
-        tffile = os.path.join(package_dir, "db", "tfs.txt")
-        tfs = pd.read_csv(tffile, header=None)[0].tolist()
-
-        tfs = expression[expression.gene.isin(tfs)]
+        tfs = expression[expression.target.isin(tfs)]
         tfs = tfs.reset_index()
         tfs = tfs.drop(columns=["index"])
-        tfs.rename(columns={"gene":"factor","target_expression":"factor_expression"},inplace=True)
-
-        # network = cartesian(expression, tfs)
-        expression['key'] = 0
-        tfs['key'] = 0
-
-        network = expression.merge(tfs, how='outer')
-
-        for col in ["factor_expression", "target_expression"]:
-            network[col + ".scale"] = minmax_scale(network[col])
-            network[col + ".rank.scale"] = minmax_scale(rankdata(network[col]))
-
-        network["source_target"] = network.factor + "_" + network.gene  
-
-        exclude_cols = [
-            "key",
-            "factor",
-            "gene",
-            "factor_expression",
-            "target_expression",
-            "factor_expression.scale",
-            "target_expression.scale",
-            # "factor_expression.rank.scale", 
-            # "target_expression.rank.scale",
-        ]
-        network = network[[c for c in network.columns if c not in exclude_cols]]
-        network = network.set_index("source_target")
-
-        bp = network.mean(axis=1)
-        bpd = pd.DataFrame(bp)
-        bpd = bpd.rename(columns={0: "binding"})
-        bpd["prob"] = minmax_scale(rankdata(bpd["binding"], method="dense"))
-
-        bpd.to_csv(outfile, sep="\t")
-
-    def run_network(self, binding, fin_expression, corrfiles, outfile):
-        # gene_bed="/home/qxu/.local/share/genomes/hg38/hg38_gffbed_piroteinCoding.bed"
-        # peak_bed="data/krt_enhancer.bed"
-        # pfmfile="../data/gimme.vertebrate.v5.1.pfm"
-        # binding="results/binding.predicted.h5"
-
-        # b=self.interaction.Interaction(genome=self.genome, gene_bed= self.gene_bed, pfmfile=self.pfmfile)
-        if not self.include_promoter and not self.include_enhancer:
-            logger.info("Create network")
-            self.create_expression_network(fin_expression, outfile)
-        elif self.include_promoter and not self.include_enhancer:
-            logger.info("Read data")
-            ddf = dd.read_csv(binding, sep="\t")[["factor", "enhancer", "binding"]]
-            tfdf, filter_bed = self.clear_peak(ddf)
-            
-            prom = self.get_promoter_dataframe(filter_bed)
+        tfs.rename(
+            columns={"target": "tf", "target_expression": "tf_expression"}, inplace=True
+        )
 
-            logger.info("Create network")
-            self.create_promoter_network(ddf, prom, fin_expression, outfile)
+        expression["key"] = 0
+        tfs["key"] = 0
```

Comment:
unused (?)

Reply:
Yeah, leftover. Not sure why they are in here. It has something to do with generating the df after that. Will check it out.